### DISCUSSION TITLE
Clarification on what target sequence means

### DISCUSSION THREAD

**Discussion by rvalieris** [milaboratory/mixcr#1508] [Q&A] [answered]
Hello,
I'm reading through the docs and I would like to confirm what "target sequence" means.

1. in the [export page](https://mixcr.com/mixcr/reference/mixcr-export/#common-fields),  "target sequence" seems to refer to the consensus sequence of the reads.
in the [mutation encoding page](https://mixcr.com/mixcr/reference/ref-mutations-encoding/),  "target sequence" seems to refer to the reference/germline gene sequence, while "query sequence" refers to the consensus sequence of the reads, is this interpretation correct ?

2. why are there multiple target sequences per clone ? should I prefer one over the other (how?) or consider both equally in down-stream analyses ? are the target sequences (of the same clone) always going to be disjointed from one another ?

4. is it possible to get the corresponding reference/germline of the target sequence like on the AIRR format ? columns like germline_alignment and  v_germline_alignment  ?

thanks for the help.

**Comment by mizraelson** [comment, marked as answer]
Hi,

`targetSequences` refers to the sequence of features used in assembling clones. For instance, if clones are assembled using `CDR3`, the `CDR3` sequence will appear in `targetSequences`. Situations where multiple `targetSequences`, separated by commas, occur include:

1) Clones assembled using more than one disjoint feature, e.g. `{FR1Begin:CDR2End}` and `{CDR3Begin:FR4End}`. In this scenario, you will observe two targetSequences, one for each region, for every clone.

2) Use of the `assembleContigs` function. In protocols like 10x, RNA-seq, and other fragmented data, clones are typically initially assembled by `CDR3` and later extended to cover as much sequence as possible. This results in different regions being covered.

Regarding the [mutation encoding page](https://mixcr.com/mixcr/reference/ref-mutations-encoding/), you are right that we refer to 'target' as the reference sequence. I apologize for any confusion caused.

To obtain a corresponding germline sequence for any gene feature, use the following parameter with `mixcr exportClones`:

`-nFeature <gene_feature> [germline]`, e.g., `-nFeature VRegion germline`.

Additionally, you can export imputed features, where uncovered parts are supplemented from the germline sequence in lowercase letters, e,g.:

`-nFeatureImputed VDJRegion`.

Please feel free to reach out if you have any further questions.

Sincerely,
Mark

**Reply by rvalieris** [reply]
thanks for the answers, just one more question:

what happens when there are multiple pre-clonotypes per group? do they become multiple target sequences?

**Reply by mizraelson** [reply]
Are you referring to the UMI/CELL_ID group?

**Reply by rvalieris** [reply]
yes, in this case UMI.

**Reply by mizraelson** [reply]
If a few reliable consensus are found within the same UMI group MiXCR tries to resolve it and make different clones from the same UMI group. But, if, for instance, there are 10 reads with different clonal sequence in a single UMI, and its not possible to isolate consensuses reliably, the whole group will be discarded.

**Reply by rvalieris** [reply]
ah so they will be separate clones, ok thanks.
